contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3617,3 +3617,40 @@ BEGIN
 END	
 $$;
 GRANT EXECUTE ON PROCEDURE sys.sp_who(IN sys.sysname, IN sys.VARCHAR(30)) TO PUBLIC;
+
+-- Change the owner of the current database.
+-- This is a wrapper around ALTER AUTHORIZATION ON DATABASE::
+CREATE OR REPLACE PROCEDURE sys.sp_changedbowner(
+	IN "@loginame" sys.sysname,
+	IN "@map"      sys.VARCHAR(5) DEFAULT NULL) -- this parameter is ignored in T-SQL
+LANGUAGE 'pltsql'
+AS $$
+BEGIN
+	DECLARE @cmd sys.NVARCHAR(300)
+	DECLARE @db  sys.sysname = DB_NAME()
+
+	-- For a NULL login name, do nothing
+	IF @loginame IS NULL
+	BEGIN
+		RETURN
+	END
+
+	IF (@db = 'master') OR (@db = 'tempdb')
+	BEGIN
+		RAISERROR('Cannot change the owner of the master or tempdb database.', 16, 1)
+		RETURN
+	END
+
+	IF SUSER_ID(@loginame) IS NULL
+	BEGIN
+		RAISERROR('Cannot find the principal ''%s'', because it does not exist or you do not have permission.', 16, 1, @loginame)
+		RETURN
+	END
+
+	-- Compose the ALTER ATHORIZATION statement:
+	SET @cmd = 'ALTER AUTHORIZATION ON DATABASE::[' + @db + '] TO [' + SUSER_NAME(SUSER_ID(@loginame)) + ']'
+	EXECUTE(@cmd)
```

Comment:
Can we avoid doing this, as it is vulnerable to SQL injection? IMO better approach would be to do it via C implementation.

Reply:
Although here I don't such risk because we are not escalating the privileges in the execution internally. But still pointing it out.

Reply:
Another perk of having it implemented in C is performance. In this impl., We access one or two catalog before we constructing and executing ALTER AUTHORIZATION equivalent syntax. I would rather implement this directly in C as a wrapper around update_db_owner (implemented in recent PR)

Reply:
I do not disagree with the principle, but this is not a performance-sensitive operation. This procedure is typically called only once manually or in a script by a DBA when a database is created, i.e. once in the lifetime of a database.

Reply:
Note that this proc is the 'old' way of doing things (i.e. Sybase-era), the reason for supporting it is just for completeness, i.e. for those folks who stick to their old habits. But basically I expect almost everyone will use ALTER AUTHORIZATION anyway. The rationale for implementing this in SQL is that this was just quicker to implement (less coding) since it's a very thin wrapper around ALTER AUTHORIZATION ON DATABASE::, and for a very limited use case.